From f8d6b66ee61864fe1206c9975ce75d4d779bf8d4 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Mon, 6 May 2024 18:14:35 +0800
Subject: [PATCH 079/236] mtk: hostapd: prevent responding to mgmt while AP MLD
 is initializing

While AP MLD is initializing, it might include incomplete information
inside its response of mgmt.
Therefore this commit prevents responding to mgmt while AP MLD is initializing

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/ieee802_11.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 4326a27c8..afaf304df 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -6689,6 +6689,13 @@ int ieee802_11_mgmt(struct hostapd_data *hapd, const u8 *buf, size_t len,
 	if (len < 24)
 		return 0;
 
+#ifdef CONFIG_IEEE80211BE
+	if (hapd->conf->mld_ap && !hapd->mld->started) {
+		wpa_printf(MSG_DEBUG, "MGMT: Drop the frame - MLD not ready");
+		return 1;
+	}
+#endif /* CONFIG_IEEE80211BE */
+
 	if (fi && fi->freq)
 		freq = fi->freq;
 	else
-- 
2.45.2

